app/commands/github/task/create_or_update.rb

```diff
@@ -0,0 +1,37 @@
+module Github
+  class Task
+    class CreateOrUpdate
+      include Mandate
+
+      initialize_with :issue_url, :attributes
+
+      def call
+        task = ::Github::Task.create_or_find_by!(issue_url: issue_url) do |t|
+          t.title = attributes[:title]
+          t.repo = attributes[:repo]
+          t.opened_at = attributes[:opened_at]
+          t.opened_by_username = attributes[:opened_by_username]
+          t.action = attributes[:action]
+          t.knowledge = attributes[:knowledge]
+          t.area = attributes[:area]
+          t.size = attributes[:size]
+          t.type = attributes[:type]
+        end
+
+        task.update!(
+          title: attributes[:title],
+          repo: attributes[:repo],
+          opened_at: attributes[:opened_at],
+          opened_by_username: attributes[:opened_by_username],
+          action: attributes[:action],
+          knowledge: attributes[:knowledge],
+          area: attributes[:area],
+          size: attributes[:size],
+          type: attributes[:type]
+        )
```

Comment:
I might make a helper for this pattern that encapsulates doing this. We have it in a few places and it's annoying to have this duplication but there's no easy way to avoid it that I know of.

Reply:
I've updated. It looks like you can just pass the hash to `update!`, which makes it very little code anyways